saleor/app/migrations/tasks/saleor3_14.py

```diff
@@ -0,0 +1,52 @@
+import graphene
+from django.db import transaction
+from django.db.models import Exists, OuterRef
+
+from .... import celeryconf
+from ....payment.models import TransactionItem
+from ...models import App
+
+# Batch size of size 100 is less than 1MB memory usage in task
+APP_BATCH_SIZE = 100
+
+# Batch size of size 1000 is about 1MB memory usage in task
+TRANSACTION_ITEM_BATCH_SIZE = 1000
+
+
+@celeryconf.app.task
+def set_identifier_for_local_apps_task():
+    apps_ids = App.objects.filter(identifier__isnull=True).values_list("pk", flat=True)[
+        :APP_BATCH_SIZE
+    ]
+    if apps_ids:
+        local_apps = App.objects.filter(id__in=apps_ids, identifier__isnull=True)
+        for app in local_apps:
+            app.identifier = graphene.Node.to_global_id("App", app.pk)
+        App.objects.bulk_update(local_apps, ["identifier"])
+
+        set_identifier_for_local_apps_task.delay()
+        set_local_app_identifier_in_transaction_item_task.delay(list(apps_ids))
+
+
+@celeryconf.app.task
+def set_local_app_identifier_in_transaction_item_task(app_ids=None):
+    if not app_ids:
+        return
+    apps = App.objects.filter(id__in=app_ids)
+    transaction_item_ids = TransactionItem.objects.filter(
+        Exists(apps.filter(id=OuterRef("app_id"))), app_identifier__isnull=True
+    ).values_list("pk", flat=True)[:TRANSACTION_ITEM_BATCH_SIZE]
+    if transaction_item_ids:
+        transaction_items = TransactionItem.objects.filter(id__in=transaction_item_ids)
```

Comment:
same, we could just operate on objects, one query less.

Reply:
Fixed. Thanks